languages/csharp/exercises/concept/exceptions/.docs/introduction.md

```diff
@@ -0,0 +1,42 @@
+Exceptions in C# provide a structured, uniform, and type-safe way of handling error conditions that occur during runtime. Proper handling of exceptions and error is important when trying to prevent application crashes.
+
+In C#, all exceptions have `System.Exception` class as their base type. It contains important properties such as `Message`, which contains a human-readable description of the reason for the exception being thrown.
+
+To signal that there should be an error in a certain part of the code, a new exception object needs to be created and then thrown,using the `throw` keyword:
+
+```csharp
+using System;
+static int Square(int number)
+{
+    if (number >= 46341)
+    {
+        throw new ArgumentException($"Argument {number} cannot be higher than 46340 as its' square doesn't fit into int type.");
+    }
+    return number * number;
+}
```

Comment:
Maybe reduce the boilerplate code here a bit to just focus on the newly introduced syntax?

```suggestion
if (number == -1)
{
    throw new ArgumentException("The number cannot be equal to -1");
}
```